poi/src/test/java/org/apache/poi/ss/formula/functions/TestSheet.java

```diff
@@ -0,0 +1,106 @@
+/* ====================================================================
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+==================================================================== */
+
+package org.apache.poi.ss.formula.functions;
+
+import org.apache.poi.hssf.usermodel.*;
+import org.apache.poi.ss.formula.OperationEvaluationContext;
+import org.apache.poi.ss.formula.eval.ErrorEval;
+import org.apache.poi.ss.formula.eval.NumberEval;
+import org.apache.poi.ss.formula.eval.StringEval;
+import org.apache.poi.ss.formula.eval.ValueEval;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.FormulaError;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+
+final class TestSheet {
+
+    private static final OperationEvaluationContext ec = new OperationEvaluationContext(null, null, 2, 0, 2, null);
+
+    @Test
+    void testCurrentSheet() {
+        ValueEval result = Sheet.instance.evaluate(new ValueEval[]{}, ec);
+        assertInstanceOf(NumberEval.class, result);
+        assertEquals(3.0, ((NumberEval) result).getNumberValue());
+    }
+
+    @Test
+    void testSheetFunctionWithRealWorkbook() throws IOException {
+        try (HSSFWorkbook wb = new HSSFWorkbook()) {
+            // Add three sheets: Sheet1, Sheet2, Sheet3
+            HSSFSheet sheet1 = wb.createSheet("Sheet1");
+            HSSFSheet sheet2 = wb.createSheet("Sheet2");
+            HSSFSheet sheet3 = wb.createSheet("Sheet3");
+
+            // Add data
+            sheet1.createRow(0).createCell(0).setCellValue(123); // A1 in Sheet1
+            sheet2.createRow(1).createCell(0).setCellValue(456); // A2 in Sheet2
+
+            HSSFFormulaEvaluator fe = new HSSFFormulaEvaluator(wb);
+
+            // Define formulas and expected results
+            String[] formulas = {
+                    "SHEET()",
+                    "SHEET(A1)",
+                    "SHEET(A1:B5)",
+                    "SHEET(Sheet2!A2)",
+                    "SHEET(\"hallo\")"
```

Comment:
how about the positive test `SHEET("Sheet2")`? We can't just test the negative case.

Reply:
Yes, that currently fails. I'll fix that on Monday.